buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-					)
-					h.Mkfile(t,
-						"build=true\ncache=true\nlaunch=true",
-						filepath.Join(appDir, "layers-A-v1", "layer.toml"),
-					)
-
-					_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-					h.AssertNotNil(t, err)
-					expected := "the launch, cache and build flags should be in the types table"
-					h.AssertStringContains(t, err.Error(), expected)
-				})
-			})
-		})
-
-		when("buildpack api = 0.2", func() {
-			it.Before(func() {
-				bpTOML.API = "0.2"
-				mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-			})
-
-			it("should convert metadata version to top level version in the buildpack plan", func() {
-				bpPlan := buildpack.Plan{
-					Entries: []buildpack.Require{
-						{
-							Name:     "some-dep",
-							Metadata: map[string]interface{}{"version": "v1"},
-						},
-					},
-				}
-
-				_, err := bpTOML.Build(bpPlan, config, mockEnv)
-				if err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-
-				testPlan(t,
-					[]buildpack.Require{
-						{
-							Name:     "some-dep",
-							Version:  "v1",
-							Metadata: map[string]interface{}{"version": "v1"},
-						},
-					},
-					filepath.Join(appDir, "build-plan-in-A-v1.toml"),
-				)
-			})
 		})
-
-		when("buildpack api < 0.5", func() {
-			it.Before(func() {
-				bpTOML.API = "0.4"
-			})
-
-			when("building succeeds", func() {
-				it.Before(func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-				})
-
-				it("should ensure the buildpack's layers dir exists and process build layers", func() {
-					h.Mkdir(t,
-						filepath.Join(layersDir, "A"),
-						filepath.Join(appDir, "layers-A-v1", "layer1"),
-						filepath.Join(appDir, "layers-A-v1", "layer2"),
-						filepath.Join(appDir, "layers-A-v1", "layer3"),
-					)
-					h.Mkfile(t, "build = true",
-						filepath.Join(appDir, "layers-A-v1", "layer1.toml"),
-						filepath.Join(appDir, "layers-A-v1", "layer3.toml"),
-					)
-					gomock.InOrder(
-						mockEnv.EXPECT().AddRootDir(filepath.Join(layersDir, "A", "layer1")),
-						mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer1", "env"), env.ActionTypePrependPath),
-						mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer1", "env.build"), env.ActionTypePrependPath),
-					)
-					gomock.InOrder(
-						mockEnv.EXPECT().AddRootDir(filepath.Join(layersDir, "A", "layer3")),
-						mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer3", "env"), env.ActionTypePrependPath),
-						mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer3", "env.build"), env.ActionTypePrependPath),
-					)
-					if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-						t.Fatalf("Unexpected error:\n%s\n", err)
-					}
-					testExists(t,
-						filepath.Join(layersDir, "A"),
-					)
-				})
-
-				it("should get bom entries and unmet requires from the output buildpack plan", func() {
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L964-L1046